### PR TITLE
refactor(webview): improve ergonomics

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
Previous interface for creating webview backend proxies was too stiff with too many niche built-ins. Things like `start` and `init` required way too much explanation.

## Solution
Drop the declarative frontend and use an abstract class instead. Derived classes act as prototype 'skeletons' from which to generate the real bindings. This removes a lot of the generic types at the cost of a small amount of runtime logic. An added benefit is that webview backends are easier to swap between a view and a panel.

Other changes:
* Renamed the entry point file to `index` to better align with common conventions
* Cleaned-up some redundant logic in all webviews
* Fixed a few styling issues
* Moved any module-level webview code to be closer together

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: resources/markdown/samReadme.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ ${COMPANYNAME} Lambda functions not defined in the [`template.yaml`](./template.
 
 ## Deploying Serverless Applications
 
-You can deploy a serverless application by invoking the `AWS: Deploy SAM application` command through the Command Palette or by right-clicking the Lambda node in the ${COMPANYNAME} Explorer and entering the deployment region, a valid S3 bucket from the region, and the name of a CloudFormation stack to deploy to. You can monitor your deployment's progress through the **${COMPANYNAME} Toolkit** Output Channel.
+You can deploy a serverless application by invoking the `AWS: Deploy SAM application` command through the Command Palette or by right-clicking the Lambda node in the ${COMPANYNAME} Explorer and entering the deployment region, a valid S3 bucket from the region, and the name of a CloudFormation stack to deploy to. You can monitor your deployment's progress through the **${COMPANYNAME} Toolkit\*\* Output Channel.
```
Comment (justinmk3) [inline resources/markdown/samReadme.md:22]: prettier keeps getting confused about this. Maybe we can change it to double quotes or backticks?

**Review by justinmk3 — APPROVED** [review]
Nice reorg

File: resources/markdown/samReadme.md (hunk 1 above)
Comment (JadenSimon) [reply resources/markdown/samReadme.md:22]: Oh it was prettier doing this...

Yeah I can try something.

Comment (JadenSimon) [reply resources/markdown/samReadme.md]: Neither worked. Weird.
It might have something to do with `**$`?

Comment (justinmk3) [reply resources/markdown/samReadme.md]: I'd just change the `**` to backticks or something.

**Review by justinmk3 — APPROVED** [review]

File: resources/markdown/samReadme.md (hunk 1 above)
Comment (JadenSimon) [reply resources/markdown/samReadme.md]: Backticks looks better anyway.

**Pull request merged**